### ISSUE TITLE
Bad/Illegal format when use %00(Null characters) on URL

### ISSUE THREAD

**Issue by tarunkant** [curl/curl#9089]
### I did this
I was running the following cURL command:

**Step 1**: Open a port 1234 from one terminal by running the following command:
`ncat -lvp 1234`

**Step 2**: From another terminal run the following cURL command:
`curl 'gopher://127.0.0.1:1234/_%0012'`

**Result:**
`curl: (3) URL using bad/illegal format or missing URL`

### I expected the following
Run without error

### curl/libcurl version

Tried on two versions of cURL
`curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1`
`curl 7.47.0 (arm-apple-darwin21.4.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1`

On both cURL version, I am getting the same error

### operating system

MacOS Monterey, M1 chip

**Comment by bagder** [comment]
The way I read [RFC 1436 section 4](https://datatracker.ietf.org/doc/html/rfc1436#section-4) nul bytes are not valid selectors for gopher. Are you saying they are?

**Comment by bagder** [comment]
> On both cURL version, I am getting the same error

Both of them show the same libcurl version used, so there's no surprise they show the same error...

**Comment by tarunkant** [comment]
When testing on Linux with libcurl 7.47.0, it works with %00(Null) characters too.

**Comment by bagder** [comment]
What works? Are you downloading gopher URLs successfully with nul bytes in the selectors?

**Comment by tarunkant** [comment]
Yes, using `libcurl/7.47.0`. I can run this command `curl 'gopher://127.0.0.1:1234/_%0012'` without any error.

**Comment by bagder** [comment]
**Are you downloading gopher URLs successfully with nul bytes in the selectors?**

They are not allowed in the gopher RFC. Why are you insisting that curl should allow them?

**Comment by bagder** [comment]
Rejected since 31e53584db5
